src/components/FeaturePanel/Climbing/ClimbingPanel.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import { Button, Dialog } from '@material-ui/core';
+import EditIcon from '@material-ui/icons/Edit';
+import { ClimbingView } from './ClimbingView';
+import { ClimbingContextProvider } from './contexts/climbingContext';
+import { ClimbingRoute } from './types';
+import { PanelScrollbars, PanelWrapper } from '../../utils/PanelHelpers';
+
+const EditRoutesContainer = styled.div`
+  padding: 10px;
+`;
+
+export const ClimbingPanel = () => {
+  const [imageSize, setImageSize] = useState({ width: 0, height: 0 });
+  const [isSelectedRouteEditable, setIsSelectedRouteEditable] = useState(false);
+  const [routes, setRoutes] = useState<Array<ClimbingRoute>>([]);
+  const [routeSelectedIndex, setRouteSelectedIndex] = useState<number>(null);
+  const [pointSelectedIndex, setPointSelectedIndex] = useState<number>(null);
+  const [isFullscreenDialogOpened, setIsFullscreenDialogOpened] =
+    useState<boolean>(true);
+
+  const onFullscreenDialogOpen = () => setIsFullscreenDialogOpened(true);
+  const onFullscreenDialogClose = () => setIsFullscreenDialogOpened(false);
+
+  const updateRouteOnIndex = (
+    routeIndex: number,
+    callback?: (route: ClimbingRoute) => ClimbingRoute,
+  ) => {
+    const updatedRoute = callback ? callback(routes[routeIndex]) : null;
+    setRoutes([
+      ...routes.slice(0, routeIndex),
+      ...(updatedRoute ? [updatedRoute] : []),
+      ...routes.slice(routeIndex + 1),
+    ]);
+  };
+
+  return (
+    <ClimbingContextProvider
+      value={{
+        imageSize,
+        setImageSize,
+        isSelectedRouteEditable,
+        setIsSelectedRouteEditable,
+        routes,
+        setRoutes,
+        routeSelectedIndex,
+        setRouteSelectedIndex,
+        pointSelectedIndex,
+        setPointSelectedIndex,
+        updateRouteOnIndex,
+      }}
+    >
+      {isFullscreenDialogOpened ? (
+        <Dialog
+          fullScreen
+          open={isFullscreenDialogOpened}
+          onClose={onFullscreenDialogClose}
+        >
+          <ClimbingView
+            isFullscreenDialogOpened={isFullscreenDialogOpened}
+            setIsFullscreenDialogOpened={setIsFullscreenDialogOpened}
+            isReadOnly={false}
+          />
+        </Dialog>
+      ) : (
```

Comment:
I think you are supposed to render the dialog always, and it will be hidden just by `open=false`. It is also better to have the Panel under it, so when animation begin or user resizes it, it looks better.

Also, could you make the dialog a little smaller? I think it should be really fullscreen only for mobile phones (see `useIsFullScreen()`). For laptop it could have 20px around so the users sees his context. For desktop and above, it can be fixed width?